src/settings.c

```diff
@@ -304,15 +305,16 @@ override_enable_animations (XdpImplSettings *impl,
 }
 
 static void
-sync_animations_enabled_override (XdpImplSettings *impl,
-                                  ShellIntrospect *shell_introspect)
+sync_animations_enabled_override (XdpImplSettings *impl)
 {
   gboolean new_enable_animations;
 
   if (!shell_introspect_are_animations_enabled (shell_introspect,
                                                 &new_enable_animations))
-
-    return;
+    {
+      SettingsBundle *bundle = g_hash_table_lookup (settings, "org.gnome.desktop.interface");
+      new_enable_animations = g_settings_get_boolean (bundle->settings, "enable-animations");
+    }
 
   override_enable_animations (impl, new_enable_animations);
```

Comment:
As it doesn't always "override" the gsetting, I think we can rename this to just `set_enable_animations()`